tools/dockerfile/interoptest/grpc_interop_go/build_interop.sh

```diff
@@ -16,18 +16,13 @@
 # Builds Go interop server and client in a base image.
 set -e
 
-# Clone just the grpc-go source code without any dependencies.
-# We are cloning from a local git repo that contains the right revision
-# to test instead of using "go get" to download from Github directly.
-git clone --recursive /var/local/jenkins/grpc-go src/google.golang.org/grpc
-
-# Get all gRPC Go dependencies
-(cd src/google.golang.org/grpc && make deps && make testdeps)
-
 # copy service account keys if available
 cp -r /var/local/jenkins/service_account $HOME || true
 
-# Build the interop client and server
-(cd src/google.golang.org/grpc/interop/client && go install)
-(cd src/google.golang.org/grpc/interop/server && go install)
-  
+if [[ -z "$GOPATH" ]]; then
+    GOPATH="$(go env GOPATH)"
+fi
+
+# Download the interop server and client at head.
+GO111MODULE=on go get google.golang.org/grpc/interop/client@HEAD \
```

Comment:
I'm not an expert on go build system, but it seems that by doing this you're always getting go interop client from 
https://github.com/grpc/grpc-go's  HEAD of master branch - which is wrong.

The interop tests are designed in a way to allows testing arbitrary version of grpc-go (e.g. in case you wanted to test an unsubmitted change locally).

On the CI, it works like this:
- the CI job checks out the "right" version of the grpc-go repository (which in this case is upstream/master, but that's just a coincidence) into a local directory and makes it available to the docker container via a mounted volume (/var/local/jenkins/)
- the interop job then tests that exact version of grpc-go by building it inside the docker container.

This setup allows you to run run_interop_tests.py locally and run interop tests on your local working copy.
All the languages are setup this way and there is no reason for go not to follow the same pattern.

So it's fine to change the go interop build, but the source code being tested needs to come from /var/local/jenkins/grpc-go. Also let's keep the comment that explains what's being done.